2019-10-28  Preksha Uttamkumar Patel  <classael@lnxsrv09.seas.ucla.edu>

            * timespec_n.h (timespec_cmp):  timespec_cmp was modified to return
            -1,0,1 instead of the difference between the two times in seconds.
            The difference resulted in an overflow which led to unexpected results.

--- lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec_n.h	2019-10-26 17:20:04.240062000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec != b.tv_sec)
+    {
+      return a.tv_sec > b.tv_sec ? 1 : -1;
+    }
+  else if (a.tv_nsec != b.tv_nsec)
+    {
+      return a.tv_nsec > b.tv_nsec ? 1 : -1;
+    }
+  return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
